packages/signer-cli/src/cmdSign.ts

```diff
@@ -13,7 +13,17 @@ const curvePrefixes: {[key in Curves]: [number] } = {
   sr25519: [1]
 };
 
+function isHexSeed (seed: string): boolean {
+  return isHex(seed) && seed.length === 66;
+}
+
+function rawValidate (seed: string): boolean {
+  return ((seed.length > 0) && (seed.length <= 32)) || isHexSeed(seed);
+}
+
 export default async function cmdSign (_: string, seed: string, type: Curves, [payload]: string[]): Promise<void> {
+  assert(rawValidate(seed), 'Invalid seed provided. Please check your input and try again.');
```

Comment:
This won't work for a full suri, i.e. `<mnemonic>//<hard>/<soft>///<password>` Since we use the `createFromUri` below, anything valid is allowed. 

Compared to the apps UI t is more complex - the apps UI had seed split from suri, here they are not, the "seed" input would be the full seed (hex/mnemonic/other) together with the derivation path.